flaml/data.py

```diff
@@ -253,13 +267,24 @@ def transform(self, X):
         X = X.copy()
         if isinstance(X, pd.DataFrame):
             cat_columns, num_columns, datetime_columns = self._cat_columns, \
-                self._num_columns, self._datetime_columns
-            X = X[cat_columns + num_columns].copy()
+                                                         self._num_columns, self._datetime_columns
             if datetime_columns:
-                for dt_column in datetime_columns:
-                    X[dt_column] = X[dt_column].map(datetime.toordinal)
+                for column in datetime_columns:
+                    tmp_dt = X[column].dt
+                    new_columns_dict = {f'year_{column}':      tmp_dt.year, f'month_{column}': tmp_dt.month,
+                                        f'day_{column}':       tmp_dt.day, f'hour_{column}': tmp_dt.hour,
+                                        f'minute_{column}':    tmp_dt.minute, f'second_{column}': tmp_dt.second,
+                                        f'dayofweek_{column}': tmp_dt.dayofweek,
+                                        f'dayofyear_{column}': tmp_dt.dayofyear,
+                                        f'quarter_{column}':   tmp_dt.quarter}
+                    for new_col_name in new_columns_dict.keys():
+                        if new_col_name in num_columns and new_col_name not in X.columns:
+                            X[new_col_name] = new_columns_dict.get(new_col_name)
+                    X[column] = X[column].map(datetime.toordinal)
+                    del tmp_dt
+            X = X[cat_columns + num_columns].copy()
             for column in cat_columns:
-                # print(column, X[column].dtype.name)
+                # print(column, X[column].dtype.name
```

Comment:
remove this line

Reply:
Sorry, i've run "coverage run -m pytest test" locally and all tests have passed. If you can tell me what types of lint problems must be solved, i can try.

Reply:
Thanks. This check fails: https://github.com/microsoft/FLAML/pull/89/checks?check_run_id=2639851684

First, make the test pass. Second, fix the lint issues in the lint section:
.\flaml\data.py:270:58: E126 continuation line over-indented for hanging indent
.\flaml\data.py:275:57: E241 multiple spaces after ':'
.\test\test_automl.py:182:9: E122 continuation line missing indentation or outdented
.\test\test_automl.py:236:9: E303 too many blank lines (2)
.\test\test_automl.py:251:5: E303 too many blank lines (2)

Reply:
@gianpDomiziani I will check in a patch that might fix the test. You can focus on the lint issues now.

Reply:
ok, i can't see the error, because locally all tests passed. Why does this happen? anyway, the lint issue should be solved now.

Reply:
Ok, the lint issues should be fixed now. Anyway, i can't realize why locally all tests passed.

Reply:
Thanks. There are still 3 lint issues left:
./test/test_automl.py:248:5: E303 too many blank lines (2)
./flaml/data.py:270:37: E126 continuation line over-indented for hanging indent
./flaml/data.py:275:57: E241 multiple spaces after ':'

Reply:
one last lint issue:
./flaml/data.py:270:21: E126 continuation line over-indented for hanging indent
I'll approve after that is fixed. Thanks.

Reply:
also please change version number to 0.4.1